### DISCUSSION TITLE
How to use inav with BRUSHED protocol and brushed motors

### DISCUSSION THREAD

**Discussion by alccrl** [iNavFlight/inav#7497] [Q&A] [answered]
Hi,
I am searching the Internet about using Inav with brushed motors, but I cannot find any useful info.
I found some bh_heli sw branch for 32bit bh_heli escs that should pilot brushed motors, but nothing else.
Since I own two very good drones with brushed motors and I would like to install an inav FC on them, I am searching info about "making the point" on this subject.

What means in the output tab BRUSHED protocol and what is is the hardware stuff and procedure to put everything to work ?
Thanks in advance ....

Alfredo

**Comment by alccrl** [comment]
After long time ..
No-one seems to be interested in this ...
Sigh !!
I think I'll have to experment by myself, and look at the code ....

**Comment by alccrl** [comment, marked as answer]
I had to wait for disassebling of a brushed quad, to answer this question.
With an oscilloscope I studied the driving waveform of the original firmware to the motors, and I realized that in practice the 4 rectifiers act as brushed ESCs.
Hence, with a F4 common controller, I chosed "BRUSHED" protocol, and, that it, motors work...

In conclusion, if You make your own brushed escs, or buy them, or flash custom brushed firmware on brushless ESCs, the result is the same, in Inav or Betaflight, it is sufficient to configure the protocol as "BRUSHED" and configure minimum throttle, or the like.

At this point I do not understand what "brushed flight controller" means, since the protocol is ESC dependent, not FC dependent.
Any FC with inav or betaflight can drive brushed Escs.

Am I wrong -:) ?

Alfredo

**Comment by Jetrell** [comment]
@alccrl I haven't used a brushed ESC for two decades. But from what I remember. They just use a 1000 - 2000uS pulse, like a servo, to drive them.
So you should be able to select `Standard` ESC protocol, under the Outputs tab.

**Reply by alccrl** [reply]
Yes, i did not check waveform detail, but I supposed being something like that.
Thanks

**Comment by DzikuVx** [comment]
Brushes FC is a FC that has MOSFETs for driving the motors built in. Any FC with broken out  outputs can be used with brushes ESCs. 
Brushed FC has just built in brushed ESC

**Reply by alccrl** [reply]
Yes,
.. it took me a while ..to realize that rectifiers with mosfets act as brushed escs .. :-) so, having them, a regular controller was enough ...or I should have bought a "brushed" controller...
Thanks

**Comment by alccrl** [comment]
Since I succeded,  after a few difficulties, in "creating" a brushed quad (issue #8642), I can assert that it has been very interesting to study brushed motors and brushed escs, in spite of their declared "obsolescence".
To be honest, my brushed escs are basically rectifiers with mosfets and a capacitor, not actually real escs, and I do not know how much are they trustable...
... anyway...
transforming a Syma x8c (more or less Helicute H809S-gps is same mechanics as Syma x8c) in a fpv quad is a very funny task.
Just throw away all original electronics and install h7 controller, pdb, gps, mag, baro, fpv camera, vtx, osd, Inav, etc... etc ... :-).
You can also plug a 4k camera and/or a 3d gimbal (I tested STORM32) under the quad (there is a standard gopro hook).

How much fly time ? => 3min - 7min with 2s LiHv 3900mah.

Could be worse ...

Alfredo
